### ISSUE TITLE
Loop overwriting a git repository

### ISSUE THREAD

**Issue by vsychov** [argoproj-labs/argocd-image-updater#227]
Hi there,

**Describe the bug**
I found bug with loop overwriting in git repository, here example of commits:
<img width="533" src="https://user-images.githubusercontent.com/2186303/125819248-97baad47-02c5-49c5-a1ad-544190497baa.png">

Inside commits looks like this:
<img width="533" src="https://user-images.githubusercontent.com/2186303/125819700-f0122cb1-a702-4ab5-a9ae-a9aa82946c13.png">

and this:
<img width="533" alt="Снимок экрана 2021-07-15 в 18 02 11" src="https://user-images.githubusercontent.com/2186303/125819879-6d120d41-1ffe-425e-89a5-c7bf18b1acd9.png">

**To Reproduce**
Create application:

```yaml
apiVersion: argoproj.io/v1alpha1
kind: Application
metadata:
  namespace: default
  name: example-app
  annotations:
    #-- shared
    argocd-image-updater.argoproj.io/image-list: backend=example.com/registry/backend,frontend=example.com/registry/frontend
    argocd-image-updater.argoproj.io/write-back-method: git
    argocd-image-updater.argoproj.io/git-branch: master
    #-- backend
    argocd-image-updater.argoproj.io/backend.update-strategy: latest
    argocd-image-updater.argoproj.io/backend.allow-tags: regexp:master_
    argocd-image-updater.argoproj.io/backend.helm.image-name: component.app.imageUrl
    argocd-image-updater.argoproj.io/backend.helm.image-tag: component.app.version
    #-- frontend
    argocd-image-updater.argoproj.io/frontend.update-strategy: latest
    argocd-image-updater.argoproj.io/frontend.allow-tags: regexp:master_
    argocd-image-updater.argoproj.io/frontend.helm.image-name: component.front.imageUrl
    argocd-image-updater.argoproj.io/frontend.helm.image-tag: component.front.version

  finalizers:
    - resources-finalizer.argocd.argoproj.io
spec:
  project: default
  source:
    repoURL: 'git@example.com:gitops/charts/my-super-app.git'
    path: .
    targetRevision: master
  destination:
    server: 'https://kubernetes.default.svc'
    namespace: default
  syncPolicy: 
    automated:
      prune: true
      selfHeal: true

```

**Expected behavior**

**Additional context**

**Version**
0.9.5

**Logs**
Please paste any relevant logs here

**Comment by jannfis** [comment]
Is this still reproducible with v0.11.0?

**Comment by vsychov** [comment]
Looks like it was fixed before or in v0.10.3
